tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/javainterop/basic/InstanceMethodTest.java

```diff
@@ -266,6 +267,23 @@ public void testGetCurrentModule() {
         BRunUtil.invoke(result, "testGetCurrentModule", args);
     }
 
+    @Test(dataProvider = "unionWithErrorTestFunctions")
+    public void testUnionWithErrorReturnArrays(String function) {
+        InstanceMethods testIns = new InstanceMethods();
+        BValue[] args = new BValue[1];
+        args[0] = new BHandleValue(testIns);
+        BRunUtil.invoke(result, function, args);
+    }
+
+    @DataProvider(name = "unionWithErrorTestFunctions")
+    public Object[][] unionWithErrorTestFunctions() {
+        return new Object[][] {
```

Comment:
Instead of an `Object[][]`, shall we return a `String[]` as we only use this to pass function names?

Reply:
Yes, we can use `String[]` as return type in this case. Addressed with https://github.com/ballerina-platform/ballerina-lang/pull/31288/commits/9a572075396e3bb39a8ea4c8d41cb107dcc44c78